lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs

```diff
@@ -27,23 +27,139 @@ namespace Org.Apache.REEF.IMRU.API
     /// <seealso cref="IMRUJobDefinitionBuilder" />
     public sealed class IMRUJobDefinition
     {
-        private readonly IConfiguration _configuration;
         private readonly string _jobName;
+        private readonly IConfiguration _mapFunctionConfiguration;
+        private readonly IConfiguration _mapInputCodecConfiguration;
+        private readonly IConfiguration _updateFunctionCodecsConfiguration;
+        private readonly IConfiguration _reduceFunctionConfiguration;
+        private readonly IConfiguration _updateFunctionConfiguration;
+        private readonly IConfiguration _mapOutputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _mapInputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _partitionedDatasetConfiguration;
+        private readonly int _numberOfMappers;
 
-        internal IMRUJobDefinition(IConfiguration configuration, string jobName)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mapFunctionConfiguration">Map function configuration</param>
+        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
+        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update 
+        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
+        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
+        /// <param name="updateFunctionConfiguration">Update function configuration</param>
+        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapOutput</param>
+        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapInput</param>
+        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned 
+        /// dataset</param>
+        /// <param name="numberOfMappers">Number of mappers</param>
+        /// <param name="jobName">Job name</param>
+        internal IMRUJobDefinition(
```

Comment:
Consider the case where both codecs of TMapInput and TMapOutput have some
base codec configuration which is same. For example, in lbfgs both
TMapinput and TMapOutput are going to have same vector codec configuration.
Isn't it going to create a conflict.

Dhruv
On Aug 3, 2015 17:16, "Markus Weimer" <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs
> <https://github.com/apache/incubator-reef/pull/334#discussion_r36146231>:
>
> > +        /// </summary>
> > +        /// <param name="mapFunctionConfiguration">Map function configuration</param>
> > +        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
> > +        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update
> > +        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
> > +        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
> > +        /// <param name="updateFunctionConfiguration">Update function configuration</param>
> > +        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapOutput</param>
> > +        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapInput</param>
> > +        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned
> > +        /// dataset</param>
> > +        /// <param name="numberOfMappers">Number of mappers</param>
> > +        /// <param name="jobName">Job name</param>
> > +        internal IMRUJobDefinition(
>
> How would there be duplications, if the Codec implementations are bound
> to NamedParameters?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/334/files#r36146231>.
>